### ISSUE TITLE
Zig fmt destoys anonymous error sets containing doc comments

### ISSUE THREAD

**Issue by kj4tmp** [ziglang/zig#22326]
### Zig Version

0.14.0-dev.2540+f857bf72e

### Steps to Reproduce and Observed Behavior

Create file test.zig

```zig
pub fn EncodeError(comptime T: type) type {
    if (T == u8) {
        return error{ NoSpaceLeft,
        /// this doc comment also crashes ZLS (see stack trace)
        SliceLenTooLarge };
    } else {
        return error{NoSpaceLeft};
    }
}

test {
    _ = EncodeError(void);
}
```

Run `zig fmt test.zig`

Observe destroyed test.zig:

```zig
pub fn EncodeError(comptime T: type) type {
    if (T == u8) {
        return error{ NoSpaceLeft, /// this doc comment also crashes ZLS (see stack trace) SliceLenTooLarge };
    } else {
        return error{NoSpaceLeft};
    }
}

test {
    _ = EncodeError(void);
}
```

zls stack trace:

```
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:775:29: 0x11fac64 in renderExpression (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:1969:41: 0x12717f1 in finishRenderBlock (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:393:35: 0x11fa0f5 in renderExpression (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:1362:33: 0x127226c in renderThenElse (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:1322:23: 0x1243759 in renderWhile (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:1252:23: 0x1243b7b in renderIf (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:879:27: 0x11f8615 in renderExpression (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:1969:41: 0x12717f1 in finishRenderBlock (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:393:35: 0x11fa0f5 in renderExpression (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:249:40: 0x12459e7 in renderMember (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/zig/render.zig:125:21: 0x10e92c0 in sendRequestSync__anon_22950 (zls)
/home/runner/work/zls/zls/src/Server.zig:2084:58: 0x10bb343 in processMessage (zls)
/home/runner/work/zls/zls/src/Server.zig:2098:33: 0x108e4c9 in processJob (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/Thread/Pool.zig:115:39: 0x10a22dc in runFn (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/Thread/Pool.zig:291:32: 0x109f32c in entryFn (zls)
/opt/hostedtoolcache/zig/0.14.0-dev.2563/x64/lib/std/os/linux/x86_64.zig:104:5: 0x10b6cf1 in clone (zls)
???:?:?: 0x0 in ??? (???)
```

### Expected Behavior

Likely no change to file.

The file compiles successfully with `zig test test.zig`.

**Comment by Rexicon226** [comment]
<details>

<summary>Stack Trace</summary>

```
thread 6108124 panic: reached unreachable code
/Users/david/Code/zig/lib/std/zig/render.zig:790:41: 0x10b4c496f in renderExpression (zig)
                        .doc_comment => unreachable, // TODO
                                        ^
/Users/david/Code/zig/lib/std/zig/render.zig:721:37: 0x10b4c3bc7 in renderExpression (zig)
                try renderExpression(r, datas[node].lhs, space);
                                    ^
/Users/david/Code/zig/lib/std/zig/render.zig:1969:41: 0x10bc1a693 in finishRenderBlock (zig)
            else => try renderExpression(r, stmt, .semicolon),
                                        ^
/Users/david/Code/zig/lib/std/zig/render.zig:1947:29: 0x10b84d6d3 in renderBlock (zig)
    return finishRenderBlock(r, block_node, statements, space);
                            ^
/Users/david/Code/zig/lib/std/zig/render.zig:393:35: 0x10b4c5163 in renderExpression (zig)
                return renderBlock(r, node, statements[0..1], space);
                                  ^
/Users/david/Code/zig/lib/std/zig/render.zig:1362:33: 0x10bc1caab in renderThenElse (zig)
            try renderExpression(r, then_expr, .space);
                                ^
/Users/david/Code/zig/lib/std/zig/render.zig:1322:23: 0x10b85a9eb in renderWhile (zig)
    try renderThenElse(
                      ^
/Users/david/Code/zig/lib/std/zig/render.zig:1252:23: 0x10b85b997 in renderIf (zig)
    return renderWhile(r, .{
                      ^
/Users/david/Code/zig/lib/std/zig/render.zig:879:27: 0x10b4c740b in renderExpression (zig)
        => return renderIf(r, tree.fullIf(node).?, space),
                          ^
/Users/david/Code/zig/lib/std/zig/render.zig:1969:41: 0x10bc1a693 in finishRenderBlock (zig)
            else => try renderExpression(r, stmt, .semicolon),
                                        ^
/Users/david/Code/zig/lib/std/zig/render.zig:1947:29: 0x10b84d6d3 in renderBlock (zig)
    return finishRenderBlock(r, block_node, statements, space);
                            ^
/Users/david/Code/zig/lib/std/zig/render.zig:393:35: 0x10b4c5163 in renderExpression (zig)
                return renderBlock(r, node, statements[0..1], space);
                                  ^
/Users/david/Code/zig/lib/std/zig/render.zig:249:40: 0x10b85e863 in renderMember (zig)
                return renderExpression(r, body_node, space);
                                       ^
/Users/david/Code/zig/lib/std/zig/render.zig:125:21: 0x10b4c76f3 in renderMembers (zig)
    try renderMember(r, container, members[0], .newline);
                    ^
/Users/david/Code/zig/lib/std/zig/render.zig:110:26: 0x10b24c19f in renderTree (zig)
        try renderMembers(&r, tree.rootDecls());
                         ^
/Users/david/Code/zig/lib/std/zig/Ast.zig:127:46: 0x10b0e2cab in renderToArrayList (zig)
    return @import("./render.zig").renderTree(buffer, tree, fixups);
                                             ^
```
</details>

**Comment by Rexicon226** [comment]
This issue is fixed by #20116
I'd also call this a duplicate of #20108
